src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingSlider/StakingSlider.tsx

```diff
@@ -0,0 +1,81 @@
+import React, { useRef } from 'react';
+import { useStakingSlider } from '~hooks';
+
+import {
+  SliderAnnotation,
+  SliderDescription,
+  StakingValidationMessage,
+} from './SliderMessages';
+
+import StakingWidgetSlider from './StakingWidgetSlider';
+import {
+  SomeSliderAnnotationProps,
+  SomeStakingValidationProps,
+  SomeStakingWidgetSliderProps,
+} from './types';
+
+const displayName =
+  'common.ColonyActions.ActionDetailsPage.DefaultMotion.StakingWidget.StakingSlider';
+
+export interface StakingSliderProps {
+  sliderAnnotationProps: SomeSliderAnnotationProps;
+  stakingWidgetSliderProps: SomeStakingWidgetSliderProps;
+  stakingValidationMessageProps: SomeStakingValidationProps;
+  mutableRef?: ReturnType<typeof useRef>;
+}
+
+const StakingSlider = ({
```

Comment:
A note on this component. 

It gets used both in the StakingWidget, and in the RaiseObjection modal (introduced in https://github.com/JoinColony/colonyCDapp/pull/301).

Dialogs don't have access to context unless you wrap the dialog provider with the provider you intend to access from the dialog (which essentially broadcasts to the entire app, like we do with `AppContext`). 

That doesn't feel like the right choice for the `StakingWidgetProvider`. It only needs to wrap the staking widget (not the whole app). So, in order to get all the necessary data to the `StakingSlider` when it's used in the Dialog (and thus outside of the `StakingWidgetProvider`, I'm passing props. A lot of them. 

I've done my best to hide this as much as possible so it doesn't look like hefty prop-drilling is taking place... but it is.